main/end-to-end-tests/cypress/integration/workspaces.spec.js

```diff
@@ -102,4 +145,39 @@ describe('Launch a new sagemaker workspace', () => {
       .parent()
       .contains('PENDING', { timeout: 600000 });
   };
+
+  const checkInstanceTypeInSelectionCard = workspaceParam => {
+    navigateToConfigurationCard(workspaceParam);
+
+    cy.get('[data-testid=configuration-card]').contains('Instance Type');
+  };
+
+  const navigateToConfigurationCard = workspaceParam => {
+    navigateToWorkspaces();
+
+    cy.get('[data-testid=workspaces]');
+
+    // Click create new workspace button
+    cy.get('button[data-testid=create-workspace]').click({ force: true });
+
+    // Select the type of environment you want to launch
+    cy.get('[data-testid=env-type-card]')
+      .contains(workspaceParam.workspaceTypeName)
+      .click();
+
+    // Click next
+    cy.get('button')
+      .contains('Next')
+      .click();
+  };
+
+  const exitConfiguration = () => {
```

Comment:
Does the code still work if we don't call `exitConfiguration`. At the end of the test, we don't have to reset it to a previous state. Each new test should `navigateToWorkspace` by itself and start from there. Therefore, we don't need to exit the configuration page.

Reply:
Nope. This was what caused the "side effect" I told you about when we talked on Friday. Since the test is mimicking creating a workspace without actually clicking create workspace, clicking the workspaces icon on the left menu (which is what `navigateToWorkspace` does) doesn't exit the process from the last test. So, this is why it was getting stuck and timing out. Although, could I possibly combine this check during the actual creation of workspaces in `launchWorkspace`? Or is that combining the tests of too many components into one case? I just fear it wouldn't be clear why it failed if we combine too many test cases.

Reply:
That's a good idea, let's do this check when we're launching the test, since launching a test requires going pass the configuration screen anyways.

In Cypress it's ok to have larger tests with more assertions.
[Link here](https://docs.cypress.io/guides/references/best-practices#Creating-tiny-tests-with-a-single-assertion)

Reply:
Doing all ^^ this worked! Now, there are only 3 E2E tests but they are pretty comprehensive!